runner/runners/polling.go

```diff
@@ -33,19 +33,19 @@ func (r *PollingStatusQuerier) QueryNow(q runner.Query) ([]runner.RunStatus, err
 
 // Query returns all RunStatus'es matching q, waiting as described by w
 func (r *PollingStatusQuerier) Query(q runner.Query, wait runner.Wait) ([]runner.RunStatus, error) {
-	if wait.Timeout == 0 {
-		return r.del.QueryNow(q)
-	}
 	end := time.Now().Add(wait.Timeout)
-	for time.Now().Before(end) {
-		st, err := r.del.QueryNow(q)
-		if err != nil || len(st) > 0 {
-			return st, err
+	for {
```

Comment:
Do we need the switch? Can we just use the switch compound conditional for the for loop (and leave the return nil/nil outside of it)?